### PR TITLE
Add support for some new WASI targets:

### PR DESCRIPTION
**Pull request by alexcrichton**
* `wasm32-wasip1`
* `wasm32-wasip2`
* `wasm32-wasip1-threads`

### CODE REVIEW

**Comment by bjorn3** [comment]
Gentle ping. https://github.com/rust-lang/miri/issues/722#issuecomment-1911016180 needs this to compile cg_clif for wasm32-wasi-preview1-threads.

**Comment by sunfishcode** [comment]
Following [this discussion](https://github.com/rust-lang/rust/pull/119616#issuecomment-1934934518), we should change these names to `wasm32-wasip1` and `wasm32-wasip2`.

**Comment by alexcrichton** [comment]
Oops yes, updated now!

**Comment by sunfishcode** [comment]
This is now released in target-lexicon 0.12.14.

File: src/targets.rs [hunk 1]
```diff
@@ -1685,6 +1694,9 @@ mod tests {
             "wasm32-unknown-emscripten",
             "wasm32-unknown-unknown",
             "wasm32-wasi",
+            "wasm32-wasip1-threads",
```
Comment (bjorn3) [inline src/targets.rs:1697]: wasm32-wasi-preview1-threads hasn't been renamed to wasm32-wasip1-threads yet and there is no open PR for this either.

Comment (alexcrichton) [reply src/targets.rs:1697]: Indeed! I was a bit forward thinking adding this in, so I suspect it won't be used for a bit.

Comment (alexcrichton) [reply src/targets.rs:1697]: Once the wasip{1,2} targets land in rustc I'll post a follow-up to rename preview1-threads to p1-threads, but in the meantime it seems reasonable to add the target here too

**Pull request merged**